src/util/belt.rs

```diff
@@ -2,9 +2,42 @@ use crate::{
     Buffer, BufferAddress, BufferDescriptor, BufferSize, BufferUsage, BufferViewMut,
     CommandEncoder, Device, MapMode,
 };
-use futures::{future::join_all, FutureExt};
+use std::pin::Pin;
+use std::task::{self, Poll};
 use std::{future::Future, sync::mpsc};
 
+// Given a vector of futures, poll each in parallel until all are ready.
+struct Join<F> {
+    futures: Vec<Option<F>>,
```

Comment:
nit: technically I guess we could just remove items directly from `futures` instead of setting them to `None`, but either seems alright

Reply:
I thought about that, but IIUC, that constitutes a move which would violate `Pin` requirements. Am I wrong?

Reply:
Yeah I think that's right – it would probably have to be behind another pointer type, so maybe not as straightforward as the `Option` approach anyway